### PR TITLE
Added support for ISO EDTT/HCI commands and ISO HCI events

### PR DESCRIPTION
**Pull request by mtpr-ot**
The following ISO HCI events have been added:
- BT_HCI_EVT_LE_CIS_ESTABLISHED
- BT_HCI_EVT_LE_CIS_REQUEST
 
The following ISO HCI commands have been added:
- le_set_cig_parameters
- le_set_cig_parameters_test
- le_create_cis
- le_remove_cig
- le_accept_cis_request
- le_reject_cis_request
- le_setup_iso_data_path
- le_remove_iso_data_path
- le_set_host_feature

The following ISO EDTT commands have been added:
- le_iso_data_flush
- le_iso_data_ready
- le_iso_data_write
- le_iso_data_read

Signed-off-by: Morten Priess <mtpr@oticon.com>

### CODE REVIEW

**Review by thoh-ot — CHANGES_REQUESTED** [review]

File: src/components/basic_commands.py [hunk 1]
```diff
@@ -3188,3 +3215,259 @@ def le_iso_data_read(transport, idx, to):
...
+                               PhyMToS, PhySToM, BnMToS, BnSToM, to):
+
+    cCB = str(CisCount) + 'B'
+    cCH = str(CisCount) + 'H'
+
+    cmd = struct.pack('<HHHB3B3BBBHBBBB' + cCB + cCB + cCH + cCH + cCH + cCH + cCB + cCB + cCB + cCB,
+                      Commands.CMD_LE_SET_CIG_PARAMETERS_TEST_REQ, 20 + (6 * CisCount * 1) + (4 * CisCount * 2), HCICommands.BT_HCI_OP_LE_SET_CIG_PARAMETERS_TEST,
+                      CigId, toArray(SduIntervalMToS, 3), toArray(SduIntervalSToM, 3), FtMToS, FtSToM, IsoInterval, SlavesClockAccuracy,
+                      Packing, Framing, CisCount, CisId, Nse, MaxSduMToS, MaxSduSToM, MaxPduMToS, MaxPduSToM,
+                      PhyMToS, PhySToM, BnMToS, BnSToM)
```
[the remark this answers is not in the document]
Comment (mtpr-ot) [reply src/components/basic_commands.py:3272]: Ouch, didn't know that. Will try to find all the palces.

File: src/components/events.py [hunk 2]
```diff
@@ -1112,6 +1117,38 @@ def __channnelSelectionAlgorithm(self):
             handle = algorithm = 0;
         return handle, algorithm;
 
+    def __cisEstablished(self):
+        if self.__checkSize(29):
+            status, connectionHandle = struct.unpack('<BH', self.data[1:4])
+            cigSyncDelay, cisSyncDelay = struct.unpack('<3B3B', self.data[4:10])
```
[the remark this answers is not in the document]
Comment (mtpr-ot) [reply src/components/events.py:1123]: So should it be this instead:
```
cigSyncDelay = struct.unpack('<3B', self.data[4:7])
cisSyncDelay = struct.unpack('<3B', self.data[7:10])
```

File: src/components/basic_commands.py [hunk 3]
```diff
@@ -3188,3 +3215,259 @@ def le_iso_data_read(transport, idx, to):
...
+    if ( 4 != len(packet) ):
+        raise Exception("LE Accept CIS Request command failed: Response too short (Expected %i bytes got %i bytes)" % (4, len(packet)))
+
+    RespCmd, RespLen = struct.unpack('<HH', packet)
+
+    if ( RespCmd != Commands.CMD_LE_ACCEPT_CIS_REQUEST_RSP ):
+        raise Exception("LE Accept CIS Request command failed: Inappropriate command response received")
+
+    if ( RespLen != 0 ):
+        raise Exception("LE Accept CIS Request command failed: Response length field corrupted (%i)" % RespLen)
```
[the remark this answers is not in the document]
Comment (mtpr-ot) [reply src/components/basic_commands.py:3364]: So:
`if ( RespLen != 1 ):`?

[the remark this answers is not in the document]
Comment (mtpr-ot) [reply src/components/basic_commands.py:3364]: So `return 0`?

File: src/components/basic_commands.py [hunk 4]
```diff
@@ -3188,3 +3215,259 @@ def le_iso_data_read(transport, idx, to):
...
+"""
+    The HCI_LE_Remove_CIG command is used by the master's Host to remove
+    all the CISes associated with the CIG identified by CIG_ID.
+"""
+def le_remove_cig(transport, idx, CigId, to):
+
+    cmd = struct.pack('<HHHB', Commands.CMD_LE_REMOVE_CIG_REQ, 7, HCICommands.BT_HCI_OP_LE_REMOVE_CIG, CigId)
+    transport.send(idx, cmd)
+
+    packet = transport.recv(idx, 5, to)
```
[the remark this answers is not in the document]
Comment (mtpr-ot) [reply src/components/basic_commands.py:3325]: Ok, hard miss.

File: src/components/basic_commands.py [hunk 5]
```diff
@@ -3188,3 +3215,259 @@ def le_iso_data_read(transport, idx, to):
...
+    parameter array.
+"""
+def le_create_cis(transport, idx, CisCount, CisConnectionHandle, AclConnectionHandle, to):
+
+    cmd = struct.pack('<HHHB' + str(CisCount) + 'H' + str(CisCount) + 'H',
+                      Commands.CMD_LE_CREATE_CIS_REQ, 7 + (2 * CisCount * 2), HCICommands.BT_HCI_OP_LE_CREATE_CIS,
+                      CisCount, CisConnectionHandle, AclConnectionHandle)
+    transport.send(idx, cmd)
+
+    packet = transport.recv(idx, 4, to)
```
[the remark this answers is not in the document]
Comment (thoh-ot) [reply src/components/basic_commands.py:3303]: There is a 1 byte status parameter in the response packet, that needs to be received, unpacked and returned.

File: src/components/basic_commands.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (thoh-ot) [reply src/components/basic_commands.py:3366]: There is a 1 byte status parameter in the response packet, that needs to be received, unpacked and returned.

**Review by thoh-ot — CHANGES_REQUESTED** [review]

File: src/components/basic_commands.py [hunk 6]
```diff
@@ -3055,3 +3106,374 @@ def gatt_service_indicate(transport, idx, to):
...
+"""
+    The HCI_LE_Remove_ISO_Data_Path command is used to remove the input
+    and/or output data path(s) associated with a CIS or BIS identified by the
+    Connection_Handle parameter.
+"""
+def le_remove_iso_data_path(transport, idx, ConnectionHandle, DataPathDirection, to):
+    
+    ConnectionHandle &= 0x0fff
+
+    cmd = struct.pack('<HHHHB,
```
Comment (thoh-ot) [inline src/components/basic_commands.py:3437]: Missing string termination in struct.pack format

File: src/components/events.py [hunk 7]
```diff
@@ -1103,6 +1117,40 @@ def __channnelSelectionAlgorithm(self):
...
+
+            if status == 0:
+                self.__checkConnectionHandle(connectionHandle)
+                self.__checkPhy(phyMToS)
+                self.__checkPhy(phySToM)
+        else:
+            status = connectionHandle = phyMToS = phySToM = nse = bNMToS = bNSToM = fTMToS = fTSToM = maxPduMToS = maxPduSToM = isoInterval = 0
+            cigSyncDelay = cisSyncDelay = transportLatencyMToS = transportLatencySToM = None
+
+        return status, connectionHandle, toNumber(cigSyncDelay), toNumber(cisSyncDelay), toNumber(transportLatencyMToS), toNumber(transportLatencySToM),
```
Comment (thoh-ot) [inline src/components/events.py:1139]: Add a `\` to the end of the line to allow for line continuation on next line

**Review by thoh-ot — CHANGES_REQUESTED** [review]

**Comment by thoh-ot** [comment]
The new commands are missing Command Complete Event decode printing format in Event.__cceFormats__

**Comment by mtpr-ot** [comment]
Creating new clean PR

**Pull request closed without merging**